nipy/algorithms/registration/groupwise_registration.py

```diff
@@ -86,6 +86,20 @@ def guess_slice_axis_and_direction(slice_info, affine):
         slice_direction = int(slice_info[1])
     return slice_axis, slice_direction
 
+def tr_from_header(images):
+    if isinstance(images, list):
+        no_init_value = True
+        for image in images:
+            tr = image.header.get_zooms()[-1]
+            if no_init_value:
+                images_tr = tr
+                no_init_value = False
+            if tr != images_tr:
+                raise ValueError('TR inconsistent between images.')
+        else:
+            return images_tr
```

Comment:
The `else` is a bit confusing here, you usually use this to run something when there was no `break` in the for loop.  Of course there isn't here, but an experienced programmer will look for one.  How about something like:

```
def tr_from_header(images):
    """ A good docstring """
    if not isinstance(images, list):
        images = [images]
    images_tr = None
    for image in images:
        tr = image.header.get_zooms()[3]
        if images_tr is None:
            images_tr = tr
        if tr != images_tr:
            raise ValueError('TR inconsistent between images.')
    return images_tr
```